src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,208 @@
+        } else {
+            succ = node(index);
+            pred = succ.prev;
+        }
+
+        for (Object o : a) {
+            Node<T> newNode = new Node<T>(pred, (T) o, null);
+            if (pred == null) {
+                first = newNode;
+            } else {
+                pred.next = newNode;
+            }
+            pred = newNode;
+        }
+
+        if (succ == null) {
+            last = pred;
+        } else {
+            pred.next = succ;
+            succ.prev = pred;
+        }
+
+        size += numNew;
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return node(index).item;
     }
 
     @Override
-    public T set(T value, int index) {
-        return null;
+    public T set(T element, int index) {
+        Node<T> x = node(index);
+        T oldVal = x.item;
+        x.item = element;
+        return oldVal;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = node(index);
+        unlink(node);
+        return node.item;
     }
 
     @Override
     public boolean remove(T t) {
+        if (t == null) {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (x.item == null) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        } else {
+            for (Node<T> x = first; x != null; x = x.next) {
+                if (t.equals(x.item)) {
+                    unlink(x);
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
     public int size() {
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        return false;
+        return size == 0;
+    }
+
+    private Node<T> node(int index) {
+        checkPositionIndex(index);
+
+        if (index < (size >> 1)) {
+            Node<T> x = first;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = last;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
+    }
+
+    private void checkPositionIndex(int index) {
+        if (!isPositionIndex(index)) {
+            throw new IndexOutOfBoundsException(outOfBoundsMsg(index));
+        }
+    }
+
+    private boolean isPositionIndex(int index) {
+        return (index >= 0 && index < size);
+    }
+
+    private void unlink(Node<T> x) {
+        Node<T> next = x.next;
+        Node<T> prev = x.prev;
+
+        if (prev == null) {
+            first = next;
+        } else {
+            prev.next = next;
+        }
+
+        if (next == null) {
+            last = prev;
+        } else {
+            next.prev = prev;
+            x.next = null;
+        }
+
+        size--;
+    }
+
+    private void linkLast(T e) {
```

Comment:
```suggestion
    private void linkLast(T element) {
```

Reply:
done

Reply:
@floriangeyer21 I don't see the fix here